test/unit/auth/auth.spec.ts

```diff
@@ -3052,6 +3053,506 @@ AUTH_CONFIGS.forEach((testConfig) => {
+          stubs.push(createTenantStub);
+          return (auth as Auth).createTenant(tenantOptions)
+            .then((actualTenant) => {
+              throw new Error('Unexpected success');
+            }, (error) => {
+              // Confirm underlying API called with expected parameters.
+              expect(createTenantStub).to.have.been.calledOnce.and.calledWith(tenantOptions);
+              // Confirm expected error returned.
+              expect(error).to.equal(expectedError);
+            });
+        });
+      });
+
+      describe('updateTenant()', () => {
+        const tenantId = 'tenant_id';
+        const tenantOptions: TenantOptions = {
+          displayName: 'TENANT_DISPLAY_NAME',
+          emailSignInConfig: {
+            enabled: true,
+            passwordRequired: true,
+          },
+        };
+        const serverResponse: TenantServerResponse = {
+          name: 'projects/project_id/tenants/tenant_id',
+          type: 'FULL_SERVICE',
+          displayName: 'TENANT_DISPLAY_NAME',
+          allowPasswordSignup: true,
+          enableEmailLinkSignin: false,
+        };
+        const expectedTenant = new Tenant(serverResponse);
+        const expectedError = new FirebaseAuthError(
+          AuthClientErrorCode.INTERNAL_ERROR,
+          'Unable to update the tenant provided.');
+        // Stubs used to simulate underlying API calls.
+        let stubs: sinon.SinonStub[] = [];
+        afterEach(() => {
+          _.forEach(stubs, (stub) => stub.restore());
+          stubs = [];
+        });
+
+        it('should be rejected given no tenant ID', () => {
+          return (auth as any).updateTenant(undefined, tenantOptions)
+            .should.eventually.be.rejected.and.have.property('code', 'auth/invalid-tenant-id');
+        });
+
+        it('should be rejected given an invalid tenant ID', () => {
+          const invalidTenantId = '';
+          return (auth as Auth).updateTenant(invalidTenantId, tenantOptions)
+            .then(() => {
+              throw new Error('Unexpected success');
+            })
+            .catch((error) => {
+              expect(error).to.have.property('code', 'auth/invalid-tenant-id');
+            });
+        });
+
+        it('should be rejected given no TenantOptions', () => {
+          return (auth as any).updateTenant(tenantId)
+            .should.eventually.be.rejected.and.have.property('code', 'auth/argument-error');
+        });
+
+        it('should be rejected given invalid TenantOptions', () => {
+          return (auth as Auth).updateTenant(tenantId, null)
+            .then(() => {
+              throw new Error('Unexpected success');
+            })
+            .catch((error) => {
+              expect(error).to.have.property('code', 'auth/argument-error');
+            });
+        });
+
+        it('should be rejected given TenantOptions with invalid property', () => {
+          return (auth as Auth).updateTenant(tenantId, {type: 'lightweight'})
```

Comment:
Can we make it clear what the invalid property is? Is it the key or the value that makes it invalid? Depending on the intention of the test either `{invalid: 'lightweight'}` or `{type: 'invalid'}`. If neither is appropriate, update the description of the test.

Reply:
I updated test description and added a comment here and in the `createTenant` test similar to this.